apstools/_devices/positioner_soft_done.py

```diff
@@ -84,7 +95,7 @@ def cb_readback(self, *args, **kwargs):
         """
         Called when readback changes (EPICS CA monitor event).
         """
-        diff = self.readback.get() - self._target.get()
+        diff = self.readback.get() - self.setpoint.get()
```

Comment:
This won't work well for some devices in which the setpoint changes when a ramp is used. For example, this is the case for the lakeshore 340. So, it should be ```diff = self.readback.get() - self.target.get()```

Reply:
The `setpoint` is the ultimate temperature where the controller is headed.  The `_target` is the interim required value for the temperature control loop.  (For example, a controller's PID loop will ramp `_target` as the immediate objective to calculate the following error: `readback - _target`.)  This `_target` value may be changed incrementally (ramped) by the controller for a controlled transition from current to `setpoint`.  We do not need to examine the controller's interim `_target` to determine if the controller has reached the requested temperature.

Definitely: `done = |readback - setpoint| <= tolerance`

Reply:
Currently, the Lakeshore 340 EPICS supports ramps the setpoint PV. Thus I think it's not great to call it setpoint in EPICS and target in Bluesky.

Reply:
You're saying the EPICS Lakeshore support uses `setpoint` for the interim setpoint of the ramp?
Doesn't the Lakeshore have a PV that reports where it is headed?  That's the PV to be called `setpoint` in Bluesky to be consistent with how it is used in other positioners, such as EpicsMotor.

The PTC10 has both: a setpoint where the controller is headed (that is `setpoint` in bluesky) and a setpoint for the PID loop (that would `target` in bluesky but this interim goal is not interesting to the user).

An interesting case is when `|target - setpoint| <= tolerance`, that's when we expect to be close enough that `|readback- setpoint| <= tolerance` could become `True`.

Reply:
We would use this for more controllers than the Lakeshore?